client.go

```diff
@@ -0,0 +1,92 @@
+package filetypes
+
+import (
+	"fmt"
+
+	"github.com/cloudquery/filetypes/csv"
+	"github.com/cloudquery/filetypes/json"
+)
+
+type FormatType string
+
+const (
+	FormatTypeCSV  = "csv"
+	FormatTypeJSON = "json"
+)
+
+type FileSpec struct {
+	Format FormatType `json:"format,omitempty"`
+	*csv.CSVSpec
+	*json.JSONSpec
```

Comment:
Just a question: Is the idea that this will get embedded by destination plugins inside their spec?

Reply:
Exactly, the `FileSpec` will be an attribute itself...

Reply:
So the destination spec will be something like this:

```
type Spec struct {
	Bucket     string              `json:"bucket,omitempty"`
	Path       string              `json:"path,omitempty"`
	FormatSpec *filetypes.FileSpec `json:"format_spec,omitempty"`
}
```